src/components/screens/vacation-requests-screen.tsx

```diff
@@ -282,15 +303,101 @@ const VacationRequestsScreen = () => {
     }
   };
 
+  /**
+   * Get updated vacation requests statuses
+   *
+   * @param newStatus vacation request status
+   * @param selectedRowIds selected row ids
+   * @returns updated vacation request statuses
+   */
+  const getUpdatedVacationRequestStatuses = async (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => {
+    const updatedVacationRequestStatuses: VacationRequestStatus[] = [];
+
+    await Promise.all(
+      selectedRowIds.map(async (selectedRowId) => {
+        const vacationRequestStatus = latestVacationRequestStatuses.find(
+          (vacationRequestStatus) => vacationRequestStatus.vacationRequestId === selectedRowId
+        );
+        try {
+          if (vacationRequestStatus?.id) {
+            const updatedVacationRequestStatus =
+              await vacationRequestStatusApi.updateVacationRequestStatus({
+                id: vacationRequestStatus.id,
+                statusId: vacationRequestStatus.id,
+                vacationRequestStatus: {
+                  ...vacationRequestStatus,
+                  status: newStatus
+                }
+              });
+            updatedVacationRequestStatuses.push(updatedVacationRequestStatus);
+          } else {
+            createVacationRequestStatus(newStatus, selectedRowId);
+          }
+        } catch (error) {
+          setError(`${strings.vacationRequestError.updateStatusError}, ${error}`);
+        }
+      })
+    );
+
+    return updatedVacationRequestStatuses;
+  };
+
+  /**
+   * Update vacation request statuses
+   *
+   * @param newStatus vacation request status
+   * @param selectedRowIds selected row ids
+   */
+  const updateVacationRequestStatuses = async (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => {
+    const updatedVacationRequestStatuses = await getUpdatedVacationRequestStatuses(
+      newStatus,
+      selectedRowIds
+    );
+
+    const foundVacationRequestStatuses = latestVacationRequestStatuses.map(
+      (latestVacationRequestStatus) => {
+        const foundUpdatedVacationRequestStatus = updatedVacationRequestStatuses.find(
+          (updatedVacationRequestStatus) =>
+            updatedVacationRequestStatus.id === latestVacationRequestStatus.id
+        );
+        if (
+          foundUpdatedVacationRequestStatus &&
+          latestVacationRequestStatus.id === foundUpdatedVacationRequestStatus.id
```

Comment:
Do you actually need  latestVacationRequestStatus.id === foundUpdatedVacationRequestStatus.id in the condition considering the .find that generates the foundUpdatedVacationStatus

Reply:
Doesn't seem to need it. Nice.